roles/ovirt-datacenter-cleanup/tasks/main.yml

```diff
@@ -0,0 +1,128 @@
+    fetch_nested: true
+
+- name: Show existing VMPools
+  debug:
+    var: ovirt_vm_pools
+
+- name: Cleanup VMPools
+  ovirt_vms:
+    state: absent
+    name: "{{ ovirt_item.name }}"
+    auth: "{{ ovirt_auth }}"
+  with_items: "{{ ovirt_vm_pools }}"
+  loop_control:
+    loop_var: ovirt_item
+
+# Cleanup VMs
+- name: Find existing VMs
+  ovirt_vms_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }}"
+    fetch_nested: true
+
+- name: Show existing VMs
+  debug:
+    var: ovirt_vms
+
+- name: Cleanup VMs
+  ovirt_vms:
+    state: absent
+    name: "{{ ovirt_item.name }}"
+    auth: "{{ ovirt_auth }}"
+  with_items: "{{ ovirt_vms }}"
+  loop_control:
+    loop_var: ovirt_item
+
+# Cleanup Templates
+- name: Find existing Templates
+  ovirt_templates_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }}"
+    fetch_nested: true
+
+- name: Show existing Templates
+  debug:
+    var: ovirt_templates
+
+- name: Cleanup Templates
+  ovirt_templates:
+    state: absent
+    auth: "{{ ovirt_auth }}"
+    name: "{{ ovirt_item.name }}"
+  with_items: "{{ ovirt_templates }}"
+  when: ovirt_item.name != 'Blank'
+  loop_control:
+    loop_var: ovirt_item
+
+# Cleanup Disks
+- name: Find existing Disks
+  ovirt_disks_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }}"
+    fetch_nested: true
+
+- name: Show existing Disks
+  debug:
+    var: ovirt_disks
+
+- name: Cleanup Disks
+  ovirt_disks:
+    state: absent
+    name: "{{ ovirt_item.name }}"
+    auth: "{{ ovirt_auth }}"
+  with_items: "{{ ovirt_disks }}"
+  loop_control:
+    loop_var: ovirt_item
+
+# Cleanup Storages
+- name: Find existing Storage Domains
+  ovirt_storage_domains_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }}"
+    fetch_nested: true
+
+- name: Show existing Storage Domains
+  debug:
+    var: ovirt_storage_domains
+
+- name: Cleanup Storage Domains
+  ovirt_storage_domains:
+    state: absent
+    name: "{{ ovirt_item.name }}"
+    auth: "{{ ovirt_auth }}"
+  with_items: "{{ ovirt_storage_domains }}"
+  when: ovirt_item.master == False
+  loop_control:
+    loop_var: ovirt_item
+
+- name: Put in maintainance master Storage Domain
+  ovirt_storage_domains:
+    state: maintenance
+    name: "{{ ovirt_item.name }}"
+    data_center: "{{ data_center_name }}"
+    auth: "{{ ovirt_auth }}"
+  with_items: "{{ ovirt_storage_domains }}"
+  when: ovirt_item.master == True
+  loop_control:
+    loop_var: ovirt_item
+
+- name: Cleanup Datacenter
+  ovirt_datacenter:
+    state: absent
+    name: "{{ data_center_name }}"
+    auth: "{{ ovirt_auth }}"
+
+- name: Remove master Storage Domain
+  ovirt_storage_domains:
+    state: absent
+    name: "{{ ovirt_item.name }}"
+    auth: "{{ ovirt_auth }}"
+  with_items: "{{ ovirt_storage_domains }}"
+  when: ovirt_item.master == True
+  loop_control:
+    loop_var: ovirt_item
```

Comment:
what about putting hosts into maintenance and remove them?
and external providers?

Reply:
Hosts and external providers don't belong to datacenter. I think it doesn't fit in the context of this role as we defined it. What do you think?

Reply:
Clusters are part of the data center and the hosts are part of the cluster. 
So i guess we should remove them. 
Becuase the situation where we have hosts on a cluster without a data center looks wierd for me. 

@machacekondra  what do you think?

Reply:
I would remove that as well, but if you don't need it you can add parameter `clean_clusters` or similar, so you can choose the behavior.